railties/lib/rails/commands/time/zones/zones_command.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails/commands/time/time_command"
+
+module Rails
+  module Command
+    module Time
+      class ZonesCommand < TimeCommand # :nodoc:
+        def all
+          require_time
+          build_time_zone_list ActiveSupport::TimeZone.all
+        end
+
+        def us
+          require_time
+          build_time_zone_list ActiveSupport::TimeZone.us_zones
+        end
+
+        def local
```

Comment:
Not a big deal, but if we add a `desc` for each of these commands, it would show in the help too:

```console
$ bin/rails time:zones:help
Commands:
  bin/rails time:zones [COUNTRY_OR_OFFSET]  # List all time zones. Can filter zones by two-letter country code (e.g. "US") or by UTC offse...
  bin/rails time:zones:all                  # 
  bin/rails time:zones:help [COMMAND]       # Describe available commands or one specific command
  bin/rails time:zones:local                # 
  bin/rails time:zones:us                   # 
```

Reply:
Ah, I missed copying the commented out `desc`s.
These were commented out to cut down on the tasks shown: https://github.com/rails/rails/commit/29acc17c0f3c6697f174b228a738819024f81c87

Reply:
Oh, good catch!  However, omitting `desc` won't omit the commands from the listing:

```console
$ bin/rails --help | grep time
time:zones                         List all time zones, list by two-letter country code (`bin/rails time:zones US`), or list by UTC offset...
time:zones:all
time:zones:local
time:zones:us
```

We could use Thor's `hide: true`, but I think that overriding `printing_commands` would give us the best of both worlds:

```ruby
class ZonesCommand < Base
  def self.printing_commands
    []
  end

  desc "all", "Display all time zones..."
  def all
    # ...
  end

  # ...
end
```

```console
$ bin/rails --help | grep time
time:zones                         List all time zones, list by two-letter country code (`bin/rails time:zones US`), or list by UTC offset...

$ bin/rails time:zones:help
Commands:
  bin/rails time:zones                 # List all time zones, list by two-letter country code (`bin/rails time:zones US`), or list by UTC ...
  bin/rails time:zones:all             # Display all time zones...
  bin/rails time:zones:help [COMMAND]  # Describe available commands or one specific command
  bin/rails time:zones:local           # 
  bin/rails time:zones:us              # 
```

Reply:
Yes, that seems the nicest solution.

Reply:
I've added the empty printing commands:
```bash
bin/rails --help | grep time
time:zones                         Show names of time zones recognized by the Rails TimeZone class
```

```bash
 bin/rails time:zones:help
Commands:
  bin/rails time:zones:all             # Display all time zones. ...
  bin/rails time:zones:help [COMMAND]  # Describe available commands or one specific command
  bin/rails time:zones:local           # Display names of time zones recognized by the Rails TimeZone class with the same offse...
  bin/rails time:zones:us              # Display names of US time zones recognized by the Rails TimeZone class, grouped by offs...

```
I think `bin/rails time:zones` is no longer shown because the `desc` now is `desc "zones [COUNTRY_OR_OFFSET]", "Show names of ....`.